### ISSUE TITLE
Can we use equivalantClass relations for alignment

### ISSUE THREAD

**Issue by ramonawalls** [Semantic-Observations/obs-models#22]
The W3C ref (not spec) states that "The meaning of such a class axiom is that the two class descriptions involved have the same class extension (i.e., both class extensions contain exactly the same set of individuals)" and "The use of owl:equivalentClass does not imply class equality. Class equality means that the classes have the same intensional meaning (denote the same concept). In the example above, the concept of "President of the US" is related to, but not equal to the concept of the principal resident of a certain estate. Real class equality can only be expressed with the owl:sameAs construct. As this requires treating classes as individuals, class equality can only be expressed in OWL Full."

It is unclear how this will work out if we specify two classes as equivalent if they have different axioms attached to them.

**Comment by ramonawalls** [comment]
To test this, I made a series of small ontologies with two properties (prop1, prop2) and three classes (thing1, thing2, thing3) each of which has a single individual (indiv1, indiv2, indiv3). For class 1, I asserted - either as a subClass axiom or as an equivalence axiom - that thing1 prop1 thing3, and for class 2 I asserted thing2 prop2 thing3. This would be more realistic if the objects of the properties were different, but that's not necessary for now.

The ontologies on in this repo under examples/classify_equivalantClasses/logical_test(1-4).owl. Ignore all the IRIs - not important.

Test1 and test2 assert that the individuals are of their respective types, while test3 and test4 do not. 
Test1 and test3 use subClass axioms for the properties, test2 and test4 use equivalent axioms.
In all 4 files, I asserted thing1 equivalantClass thing2
You really only need to look at test4 to see the results. The others give similar results, as expected.

Looking at test4, using either the Hermit or FACT++ reasoner in Protege, both indiv1 and indiv2 are classified as members of both thing1 and thing2.

@mbjones , does this answer your question?

**Comment by amoeba** [comment]
@mbjones and I worked on this last week and I've worked on it some more this morning. Most of the work was just getting me familiar with this problem space.

I re-created a very similar set of ontologies to yours and wrote to shell scripts that run `sparql` and `owltools` so I could have a command-line way of understanding the implications of the ways each ontology is specified. These will likely seem very redundant to your ontologies.

One thing in particular that @mbjones and I were curious about was whether all individuals of a class need to satisfy their `minCardinality` restrictions upon creation. Like here:

``` xml
    <owl:Class rdf:about="&obo;B">
        <rdfs:label xml:lang="en">B</rdfs:label>
        <rdfs:subClassOf>
            <owl:Restriction>
                <owl:onProperty rdf:resource="&obo;hasBProp"/>
                <owl:onClass rdf:resource="&obo;B"/>
                <owl:minCardinality rdf:datatype="&xsd;nonNegativeInteger">1</owl:minCardinality>
            </owl:Restriction>
        </rdfs:subClassOf>
    </owl:Class>
```

If I go on to create a named individual of type B, is it invalid until I assert that my individual has at least one `hasBProp` triple? As far as we can tell, this is totally valid and that it's due to the open world philosophy of OWL which implies that what I've described isn't invalid because the triple we are trying to require through the `minCardinality` triple could exist in another graph.

Everything I've done so far shows that equivalentClass works for what we want to do and that those classes can have restrictions like minCardinality and everything will be fine.

Kind of a separate issue, but I also looked at SPARQL queries. We'd like to be able to query for individuals of a class and get all individuals that are directly typed as that class plus any individuals that are typed to classes that are equivalentClass to the class we queried for. e.g.

``` sparql
select ?i where {
  ?i a obo:A
}
```

But there is no configuration where a basic SPARQL query returns more than just the individuals directly typed as A. You'd have to use a query like:

``` sparql
select ?i ?type where {
  ?i a/(rdfs:subClassOf|owl:equivalentClass|^owl:equivalentClass)* obo:A
}
```

So if we want to create SPARQL queries to show the alignment ontology working, we'll have to do a bit more work.
